archunit/src/main/java/com/tngtech/archunit/lang/syntax/elements/ClassesShould.java

```diff
@@ -257,6 +257,14 @@
     @PublicAPI(usage = ACCESS)
     ClassesShouldConjunction haveOnlyFinalFields();
 
+    /**
+     * Asserts that classes have only private constructors.
+     *
+     * @return A syntax element that can either be used as working rule, or to continue specifying a more complex rule
+     */
+    @PublicAPI(usage = ACCESS)
+    ClassesShouldConjunction haveOnlyPrivateConstructors();
```

Comment:
It would have been much better to expose `haveOnlyConstructorsWithModifiers` as well

Reply:
We're open for PR's :wink:
Should be easy to pipe this method to a new `haveOnlyConstructorsWithModifiers(..)` method.
Other than that you can of course already test this with the current state, just directly use the predicates / conditions and adjust them...